eth/main.cpp

```diff
@@ -1137,13 +1137,33 @@ int main(int argc, char** argv)
 			else if (c && cmd == "setblockfees")
 			{
 				iss >> blockFees;
-				gasPricer->setRefBlockFees(u256(blockFees * 1000));
+				try
+				{
+					gasPricer->setRefBlockFees(u256(blockFees * 1000));
+				}
+				catch(Overflow const& _e)
+				{
+					cout << boost::diagnostic_information(_e);
+				}
+
 				cout << "Block fees: " << blockFees << endl;
 			}
 			else if (c && cmd == "setetherprice")
 			{
 				iss >> etherPrice;
-				gasPricer->setRefPrice(u256(double(ether / 1000) / etherPrice));
+				if (etherPrice == 0)
+					cout << "ether price can not be set to zero" << endl;
+				else
+				{
+					try
+					{
+						gasPricer->setRefPrice(u256(double(ether / 1000) / etherPrice));
```

Comment:
Thats actually Gavs code ;-), but nevertheless, I also prefer to make the cast as late as necessary for accuracy. So ether / 1000 is fine, but when divided by etherPrice it needs to be a double, therefore the cast. The cast from double to u256 is also necessary.